Lib/json/__main__.py

```diff
@@ -0,0 +1,21 @@
+r"""Command-line tool to validate and pretty-print JSON
+
+Usage::
+
+    $ echo '{"json":"obj"}' | python -m json
+    {
+        "json": "obj"
+    }
+    $ echo '{ 1.2:3.4}' | python -m json
+    Expecting property name enclosed in double quotes: line 1 column 3 (char 2)
+
+"""
+import json.tool
+import sys
+
+
+if __name__ == '__main__':
+    try:
+        json.tool.main()
+    except BrokenPipeError as exc:
+        sys.exit(exc.errno)
```

Comment:
`sys.exit` just calls `SystemExit` 
```suggestion
import json.tool


if __name__ == '__main__':
    try:
        json.tool.main()
    except BrokenPipeError as exc:
        raise SystemExit(exc.errno) from None
```

Reply:
This code originated in the `json.tool` module.
Calling `sys.exit` seems to be more common than raising `SystemExit` within the standard library. I'm not sure why that is, but I assumed consistency might be valued here.

I'm fine with either use though.

Reply:
We use `raise SystemExit(e)` in `json.tool`, so I think for consistency using `SystemExit` in `json.__main__` would be nice?

A

Reply:
Interestingly, both `SystemExit` and `sys.exit` seem to be used in `json.tool`. ([1](https://github.com/python/cpython/blob/b3d2427f2280fa8dae3515036c518d74ba43ebd1/Lib/json/tool.py#L82) and [2](https://github.com/python/cpython/blob/b3d2427f2280fa8dae3515036c518d74ba43ebd1/Lib/json/tool.py#L89)).

Agreed that consistency seems useful here. Any preference toward updating both uses of `sys.exit` (the one in `json.__main__` and the one in `json.tool`) to use `SystemExit` versus updating all to use `sys.exit`?

Reply:
If it were me, I'd go for `SystemExit` out of explicitness -- doesn't hide the exception, etc. Up to you, though.